libs/datamodel/core/src/transform/ast_to_dml/lift.rs

```diff
@@ -276,7 +276,7 @@ impl<'a> LiftAstToDml<'a> {
         model.is_ignored = walker.is_ignored();
 
         model.primary_key = walker.primary_key().map(|pk| dml::PrimaryKeyDefinition {
-            name: pk.name().map(String::from),
+            name: pk.name().as_explicit().map(String::from),
```

Comment:
Does this mean there is still some logic in the QE that will generate the implicit names again for the Client 
API since they are not passed here?

Reply:
Yes. We initially wanted to do that QE side at runtime to prevent having duplicated logic, but it turns out we have no `Result` type in the whole query schema building stack and we don't want to introduce hundreds of `Result` types for this specific edge-case.